internal/clientconn/otel_propagation.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientconn
+
+import (
+	"encoding/json"
+
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// spanContextFromComment extracts OpenTelemetry tracing information from comment.
+//
+// If the comment is empty, it returns an empty span context and no error.
+func spanContextFromComment(comment string) (trace.SpanContext, error) {
+	if comment == "" {
+		return trace.SpanContext{}, nil
+	}
+
+	type TraceData struct {
+		TraceID [16]byte `json:"ferretTraceID"`
+		SpanID  [8]byte  `json:"ferretSpanID"`
+	}
```

Comment:
Let's add injecting/extracting code into `internal/util/observability` package so the format is not duplicated.
Let's also improve the format (right now, it is a JSON array, which is weird: https://go.dev/play/p/lE11AQSHbsa) and add a top-level field instead of adding prefixes to fields. After all, it is not some special FerretDB Trace ID – it is a normal OTel Trace ID, just wrapped with another object to avoid clashes.